client/my-sites/stats/hooks/use-jetpack-connection-status.ts

```diff
@@ -0,0 +1,39 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import getDefaultQueryParams from './default-query-params';
+
+interface ConnectionInfo {
+	isRegistered: boolean;
+	isUserConnected: boolean;
+	isSiteFullyConnected?: boolean;
+}
+
+function queryJetpackConnectionStatus(
+	siteId: number | null,
+	isWpcomSite = false
+): Promise< ConnectionInfo > {
+	if ( isWpcomSite ) {
```

Comment:
WPCOM sites are always connected.

Reply:
Is this also true for Atomic sites?

Reply:
Yes I believe so. Asked here just in case: p1727412713650889-slack-CDLH4C1UZ

Reply:
The feedback from the slack thread is that the connection could break. So  I changed to only assume only simple sites don't need to test the connection status in e5d51be4f4